ports/mimxrt/machine_pin.c

```diff
@@ -0,0 +1,325 @@
+STATIC mp_obj_t machine_pin_off(mp_obj_t self_in) {
+    machine_pin_obj_t *self = self_in;
+    mp_hal_pin_low(self);
+    return mp_const_none;
+}
+STATIC MP_DEFINE_CONST_FUN_OBJ_1(machine_pin_off_obj, machine_pin_off);
+
+// pin.on()
+STATIC mp_obj_t machine_pin_on(mp_obj_t self_in) {
+    machine_pin_obj_t *self = self_in;
+    mp_hal_pin_high(self);
+    return mp_const_none;
+}
+STATIC MP_DEFINE_CONST_FUN_OBJ_1(machine_pin_on_obj, machine_pin_on);
+
+// pin.value([value])
+STATIC mp_obj_t machine_pin_value(size_t n_args, const mp_obj_t *args) {
+    return machine_pin_obj_call(args[0], (n_args - 1), 0, args + 1);
+}
+STATIC MP_DEFINE_CONST_FUN_OBJ_VAR_BETWEEN(machine_pin_value_obj, 1, 2, machine_pin_value);
+
+// pin.mode()
+STATIC mp_obj_t machine_pin_mode(mp_obj_t self_in) {
+    machine_pin_obj_t *self = self_in;
+    return mp_obj_new_int(pin_get_mode(self));
+}
+STATIC MP_DEFINE_CONST_FUN_OBJ_1(machine_pin_mode_obj, machine_pin_mode);
+
+// pin.init(mode, pull, [kwargs])
+STATIC mp_obj_t machine_pin_init(size_t n_args, const mp_obj_t *args, mp_map_t *kw_args) {
+    return machine_pin_obj_init_helper(args[0], n_args - 1, args + 1, kw_args);
+}
+MP_DEFINE_CONST_FUN_OBJ_KW(machine_pin_init_obj, 1, machine_pin_init);
+
+// pin.drive([drive])
+STATIC mp_obj_t machine_pin_drive(size_t n_args, const mp_obj_t *args) {
+    machine_pin_obj_t *self = args[0];
+    uint32_t pad_config = *((volatile uint32_t *)self->configRegister);
+
+    if (n_args == 1) {
+        pad_config &= IOMUXC_SW_PAD_CTL_PAD_DSE_MASK;
+        pad_config = pad_config >> IOMUXC_SW_PAD_CTL_PAD_DSE_SHIFT;
+        return MP_OBJ_NEW_SMALL_INT(pad_config);
+    } else {
+        uint drive = MP_OBJ_SMALL_INT_VALUE(args[1]);
+
+        if (!IS_GPIO_DRIVE(drive)) {
+            mp_raise_msg_varg(&mp_type_ValueError, MP_ERROR_TEXT("invalid drive strength: %d"), drive);
+        }
+
+        // Update drive settings of pad configuration
+        pad_config &= ~IOMUXC_SW_PAD_CTL_PAD_DSE_MASK;
+        pad_config |= IOMUXC_SW_PAD_CTL_PAD_DSE(drive);
+
+        IOMUXC_SetPinConfig(0, 0, 0, 0, self->configRegister, pad_config);
+        return mp_const_none;
+    }
+}
+STATIC MP_DEFINE_CONST_FUN_OBJ_VAR_BETWEEN(machine_pin_drive_obj, 1, 2, machine_pin_drive);
+
+STATIC const mp_rom_map_elem_t machine_pin_locals_dict_table[] = {
+    // TODO: Implement class locals dictionary
+    // instance methods
+    { MP_ROM_QSTR(MP_QSTR_off),     MP_ROM_PTR(&machine_pin_off_obj) },
+    { MP_ROM_QSTR(MP_QSTR_on),      MP_ROM_PTR(&machine_pin_on_obj) },
+    { MP_ROM_QSTR(MP_QSTR_low),     MP_ROM_PTR(&machine_pin_off_obj) },
+    { MP_ROM_QSTR(MP_QSTR_high),    MP_ROM_PTR(&machine_pin_on_obj) },
+    { MP_ROM_QSTR(MP_QSTR_value),   MP_ROM_PTR(&machine_pin_value_obj) },
+    { MP_ROM_QSTR(MP_QSTR_mode),    MP_ROM_PTR(&machine_pin_mode_obj) },
+    { MP_ROM_QSTR(MP_QSTR_init),    MP_ROM_PTR(&machine_pin_init_obj) },
+    { MP_ROM_QSTR(MP_QSTR_drive),   MP_ROM_PTR(&machine_pin_drive_obj) },
```

Comment:
For now, please don't implement `mode()` or `drive()` methods.  They are not part of the core Pin API and the functionality of setting these values should be available via `init()`.

Reply:
Ok. 👌

Reply:
`drive()` and  `mode()`methods have been removed